contrib/jemalloc/include/jemalloc/internal/jemalloc_internal_inlines_c.h

```diff
@@ -31,6 +31,25 @@ unbound_ptr(tsdn_t *tsdn, void *ptr) {
 	rtree_ctx_t rtree_ctx_fallback;
 	extent_t *extent;
 
+	/*
+	 * These checks catch the most egregious attacks from an adversary
+	 * who can't manipulate the bounds of capabilities can manipulate
+	 * the offset or storage of a pointer passed to free() or realloc().
+	 *
+	 * XXX: Fruther validation of the pointer relative to the extent
+	 * metadata is required to prevent attacks where bounds are
+	 * manipulated.
+	 */
+	if (unlikely(!cheri_gettag(ptr))) {
+		malloc_write("<jemalloc>: can't unbound invalid cap\n");
+		abort();
+	}
+	if (unlikely(cheri_getoffset(ptr) != 0)) {
```

Comment:
Have we made sure we use csetboundsexact for all capabilities returned from malloc? If not this could trigger due to a bug in malloc().

Reply:
Yes unless you disable setting bounds.  I'll remove that option.

Reply:
I missed a case that had caused us to not use csetboundsexact, but in practice it was provably OK (the size was rounded in the underlying allocation).  I've now fixed the `calloc` case and switched us to csetboundsexact.